src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -738,5 +738,58 @@ public static void Uri_DoesNotLockOnString()
             Assert.True(Monitor.TryEnter(uriString, TimeSpan.FromSeconds(10)));
             Assert.False(timedOut);
         }
+
+        public static IEnumerable<object[]> FilePathHandlesNonAscii_TestData()
+        {
+            if (PlatformDetection.IsNotWindows)
```

Comment:
Does this mean that `Uri` cannot handle Unix paths on Windows and vice-versa? Does the parsing itself depends on what OS the code is running?

Reply:
> Does this mean that Uri cannot handle Unix paths on Windows and vice-versa?

Unix paths in implicit form will not be recognized on Windows.
`new Uri("/foo")` will throw on Windows and succeed on Unix.

DOS paths (`C:/foo`) are recognized the same on any platform.

> Does the parsing itself depend on what OS the code is running?

Unfortunately, yes.

Reply:
Thanks for explanation!